centrifuge.wdl

```diff
@@ -1,124 +1,201 @@
 version 1.0
 
-# Copyright Sequencing Analysis Support Core - Leiden University Medical Center 2018
+# Copyright (c) 2020 Sequencing Analysis Support Core - Leiden University Medical Center
 #
-# Tasks from centrifuge
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
 task Build {
     input {
+        Boolean disableDifferenceCover = false
         File conversionTable
         File taxonomyTree
-        File inputFasta
-        String centrifugeIndexBase
-        String? preCommand
-        String centrifugeBuildExecutable = "centrifuge-build"
-        #Boolean? c = false
-        Boolean largeIndex = false
-        Boolean noAuto = false
-        Int? bMax
-        Int? bMaxDivn
-        Boolean noDiffCover = false
-        Boolean noRef = false
-        Boolean justRef = false
-        Int? offRate
-        Int? fTabChars
-        File? nameTable
-        File? sizeTable
-        Int? seed
+        File nameTable
+        File referenceFile
+        String indexBasename = "NAME"
+        String outputPrefix
+
+        Int? offrate
+        Int? ftabChars
         Int? kmerCount
+        File? sizeTable
 
-        Int threads = 8
+        Int cores = 5
         String memory = "20G"
+        String dockerImage = "quay.io/biocontainers/centrifuge:1.0.4_beta--he860b03_3"
```

Comment:
Should we be using a beta version?

Reply:
The non beta tagged version on quay.io is from may 2018. So I thought using the most recent one would be better.

Reply:
Okay, I guess that's fine.